build.js

```diff
@@ -164,12 +164,124 @@ function buildLocale (source, locale) {
 
   // This actually executes the build and stops the internal timer after
   // completion.
-  metalsmith.build((err) => {
+  metalsmith.build((err, files) => {
     if (err) { throw err }
     console.timeEnd(`[metalsmith] build/${locale} finished`)
+    if (opts && opts.preserveLocale) {
+      buildMissing(source, locale, files)
+    }
   })
 }
 
+function buildMissing (source, locale, localeFiles) {
+  console.log(`[metalsmith] build/${locale} missing files started`)
+  console.time(`[metalsmith] build/${locale} missing files finished`)
+
+  const metalsmith = Metalsmith(__dirname)
```

Comment:
Could you please refactor this part, so we don't have the entire build chain twice?

If you want to distinguish regular/local builds, you can also use something like:

```js
function onlyLocalBuild (child) {
  if (/* whatever flag*/) {
	// return a dummy function if there's nothing to do
    return function (files, metalsmith, done) {
      done()
    }
  }
  return child
}
```

```js
.use(onlyLocalBuild(myFunction({
// ...
})))
```

Reply:
Sounds good to me!